motoman_sia50_support/urdf/sia50d_macro.xacro

```diff
@@ -0,0 +1,209 @@
+            </visual>
+            <collision>
+                <origin rpy="0 0 0" xyz="0 0 0"/>
+                <geometry>
+                    <mesh filename="package://motoman_sia50_support/meshes/sia50d/collision/link_4_u.stl"/>
+                </geometry>
+                <material name="yellow"/>
+            </collision>
+        </link>
+        <link name="${prefix}link_r">
+            <visual>
+                <origin rpy="0 0 0" xyz="0 0 0"/>
+                <geometry>
+                    <mesh filename="package://motoman_sia50_support/meshes/sia50d/visual/link_5_r.stl"/>
+                </geometry>
+                <material name="silver"/>
+            </visual>
+            <collision>
+                <origin rpy="0 0 0" xyz="0 0 0"/>
+                <geometry>
+                    <mesh filename="package://motoman_sia50_support/meshes/sia50d/collision/link_5_r.stl"/>
+                </geometry>
+                <material name="yellow"/>
+            </collision>
+        </link>
+        <link name="${prefix}link_b">
+            <visual>
+                <origin rpy="0 0 0" xyz="0 0 0"/>
+                <geometry>
+                    <mesh filename="package://motoman_sia50_support/meshes/sia50d/visual/link_6_b.stl"/>
+                </geometry>
+                <material name="blue"/>
+            </visual>
+            <collision>
+                <origin rpy="0 0 0" xyz="0 0 0"/>
+                <geometry>
+                    <mesh filename="package://motoman_sia50_support/meshes/sia50d/collision/link_6_b.stl"/>
+                </geometry>
+                <material name="yellow"/>
+            </collision>
+        </link>
+        <link name="${prefix}link_t">
+            <visual>
+                <geometry>
+                    <mesh filename="package://motoman_sia50_support/meshes/sia50d/visual/link_7_t.stl" />
+                </geometry>
+                <material name="silver"/>
+            </visual>
+            <collision>
+                <geometry>
+                    <mesh filename="package://motoman_sia50_support/meshes/sia50d/collision/link_7_t.stl" />
+                </geometry>
+                <material name="yellow"/>
+            </collision>
+        </link>
+        <link name="${prefix}tool0" />
+        <!-- end of link list -->
+        <!-- joint list -->
+        <joint name="${prefix}joint_s" type="revolute">
+            <parent link="${prefix}base_link"/>
+            <child link="${prefix}link_s"/>
+            <origin xyz="0 0 0.312000" rpy="0 0 0"/>
+            <axis xyz="0 0 1" />
+            <!-- Max Acc/Dec Torque: 4903.32, Rated Motor Torque: 18.5 -->
+            <limit lower="-3.1415" upper="3.1415" effort="18.5" velocity="2.967051" />
+        </joint>
+        <joint name="${prefix}joint_l" type="revolute">
+            <parent link="${prefix}link_s"/>
+            <child link="${prefix}link_l"/>
+            <origin xyz="0.145 0 0.228" rpy="0 0 0"/>
```

Comment:
Ideally `joint`s have translations in a single dimension only. None of the other `joint`s in your macro have this double translation.

Is it required?

Reply:
It's due to this:

![image](https://user-images.githubusercontent.com/1664508/85311203-f7069600-b4ac-11ea-9e37-9156ebc45cc1.png)

![image](https://user-images.githubusercontent.com/1664508/85311239-071e7580-b4ad-11ea-8990-8d826d24bf02.png)

Reply:
This is solvable by placing `link_s` in the same place as `link_l`. That's beneficial, as that would also be the location of the `base` frame.